### PR TITLE
add transition callbacks to Toast

### PR DESCRIPTION
**Pull request by ysshir**
Add animation callbacks to Toast.

Fixes https://github.com/react-bootstrap/react-bootstrap/issues/6152
The previous commit https://github.com/react-bootstrap/react-bootstrap/pull/6261 has not been merged.

### CODE REVIEW

File: src/Toast.tsx [hunk 1]
```diff
@@ -51,6 +53,36 @@ const propTypes = {
    */
   onClose: PropTypes.func,
 
+  /**
+   * Callback fired before the Modal transitions in
+   */
+  onEnter: PropTypes.func,
+
+  /**
+   * Callback fired as the Modal begins to transition in
```
Comment (kyletsang) [inline src/Toast.tsx:62]: Typo with "Modal"

**Comment by kyletsang** [comment]
Could you please add tests?  Thanks

**Comment by ysshir** [comment]
I wrote a test and a example.
in my local

**Comment by ysshir** [comment]
@kyletsang

Thank you for the review.

I've corrected the typo and written a test and an example. It works fine in the example, but the test failed. I noticed that the "onEntered" is not being called, but I couldn't figure out why.

Any ideas? Thanks.

**Comment by kyletsang** [comment]
The toast tests uses fake timers so you need to manually advance the clock so code within functions like `setTimeout` will trigger.  If you check the other tests, they call the tick function like `clock.tick(250);`.  You'll need to do this within your test.

**Comment by ysshir** [comment]
@kyletsang

Thank you for the advice!!
I believe the test is okay for now.

**Comment by kyletsang** [comment]
Code and tests look good.  The example currently isn't mobile friendly though.  I'm thinking we probably don't need the example at all actually, since the props are pretty straight forward.  Could you remove the example from this PR?

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks!

**Pull request merged**